aten/src/ATen/native/TensorShape.cpp

```diff
@@ -148,6 +148,39 @@ Tensor &as_strided_(Tensor& self, IntList size, IntList stride) {
   return at::as_strided_(self, size, stride, self.storage_offset());
 }
 
+Tensor narrow_copy_sparse(const Tensor& self, int64_t dim, int64_t start, int64_t length){
+  AT_CHECK(self.dim() > 0, "narrow() cannot be applied to a 0-dim tensor.");
```

Comment:
can you AT_CHECK for the range of dim, start and length here so that we can assume invariants from this point?

Reply:
ok, lifted this from the standard narrow. that what you mean?

Reply:
yes, that what I meant